scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingService.java

```diff
@@ -977,17 +978,37 @@ void pingTaskNode(RunningTaskData taskData) {
         } catch (Throwable t) {
             RunningTaskData runningTask = jobs.getRunningTask(task.getId());
             if (runningTask != null) {
-                int attempts = runningTask.increaseAndGetPingAttempts();
-                if (attempts > PASchedulerProperties.SCHEDULER_NODE_PING_ATTEMPTS.getValueAsInt()) {
-                    tlogger.info(task.getId(), "node failed", t);
-                    restartTaskOnNodeFailure(task);
+                // We try to ping the node where the task is running.
+                // We don't consider here other nodes reserved for the task, 
+                // as it is the responsibility of the task itself to manage extra nodes lifecycle 
+                // in case of complex multinodes task deployment.
+                if (!pingNode(runningTask.getNodes().get(0))) {
```

Comment:
Could you please explain why it is required to ping the running Task in this catch block? If we are in this catch block, it seems that we already have detected a connection issue with the Node. Why do you need to ping again?